filecoin-proofs/src/api/update.rs

```diff
@@ -158,7 +174,7 @@ pub fn encode_into<Tree: 'static + MerkleTreeTrait<Hasher = TreeRHasher>>(
         get_new_configs_from_t_aux_old::<Tree>(&t_aux, new_cache_path, config.nodes_count)?;
 
     let (comm_r_domain, comm_r_last_domain, comm_d_domain) =
-        EmptySectorUpdate::<Tree>::encode_into(
+        EmptySectorUpdate::<Fr, Tree::Arity, Tree::SubTreeArity, Tree::TopTreeArity>::encode_into(
```

Comment:
Note: `EmptySectorUpdate` is no longer generic over `TreeR: MerkleTreeTrait`; it is now generic over `F, U, V, W`.